src/components/Menu/MenuItem.tsx

```diff
@@ -142,6 +151,50 @@ export function MenuItem({
         )}
       </li>
     );
+  } else if (!pageNode.platforms) {
+    // This check is for the gen2 pages
+    // check if a pagenode has no platforms at all because that would mean it's a gen2 page
+    // does it make sense to have the menu component try to cover these cases or should
+    // there be a separate component for gen2 and classic?
```

Comment:
Looking for feedback on this, not sure if this is the right way to think about it or not

Reply:
I'm not seeing much of a difference between the two conditions (a query param passed into link and currentPlatform passed into menu item.  Seems like we can just combine these two conditions.

Reply:
After syncing with Jacob offline, we cleaned up some of the code for this!